src/main.cpp

```diff
@@ -5319,42 +5514,211 @@ bool static ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vR
         }
+        return true;
     }
-    else if (strCommand == NetMsgType::XPEDITEDREQUEST)
+};
+
+class ExpeditedRequestMessage : public NetworkMessage {
+public:
+    ~ExpeditedRequestMessage() {}
+
+    ExpeditedRequestMessage(CNode * const pFrom) : NetworkMessage(pFrom) {}
+
+    bool handle(CDataStream &vRecv, int64_t nTimeReceived,  std::string &strCommand)
     {
-        HandleExpeditedRequest(vRecv,pfrom);
+        HandleExpeditedRequest(vRecv, pfrom);
+        return true;
     }
-    else if (strCommand == NetMsgType::XPEDITEDBLK)
+};
+
+class ExpeditedBlockMessage : public NetworkMessage {
+public:
+    ~ExpeditedBlockMessage() {}
+
+    ExpeditedBlockMessage(CNode * const pFrom) : NetworkMessage(pFrom) {}
+
+    bool handle(CDataStream &vRecv, int64_t nTimeReceived,  std::string &strCommand)
     {
-        HandleExpeditedBlock(vRecv,pfrom);
+        HandleExpeditedBlock(vRecv, pfrom);
+        return true;
     }
-    else if (strCommand == NetMsgType::VERSION2)
+};
+
+class Version2Message : public NetworkMessage {
+public:
+    ~Version2Message() {}
+
+    Version2Message(CNode * const pFrom) : NetworkMessage(pFrom) {}
+
+    bool handle(CDataStream &vRecv, int64_t nTimeReceived,  std::string &strCommand)
     {
         // Each connection can only send one version message
         if (pfrom->addrFromPort != 0) {
-            pfrom->PushMessage(NetMsgType::REJECT, strCommand, REJECT_DUPLICATE, std::string("Duplicate version2 message"));
+            pfrom->PushMessage(NetMsgType::REJECT, strCommand, REJECT_DUPLICATE,
+                               std::string("Duplicate version2 message"));
             LOCK(cs_main);
             Misbehaving(pfrom->GetId(), 15);
             return false;
         }
 
         vRecv >> pfrom->addrFromPort;
         pfrom->PushMessage(NetMsgType::VERACK2);
+        return true;
     }
-    else if (strCommand == NetMsgType::VERACK2)
+};
+
+class VersionAck2Message : public NetworkMessage {
+public:
+    ~VersionAck2Message() {}
+
+    VersionAck2Message(CNode * const pFrom) : NetworkMessage(pFrom) {}
+
+    bool handle(CDataStream &vRecv, int64_t nTimeReceived,  std::string &strCommand)
     {
         CheckAndRequestExpeditedBlocks(pfrom);
+        return true;
+    }
+};
+
+bool static ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, int64_t nTimeReceived)
+{
+    const CChainParams& chainparams = Params();
+    RandAddSeedPerfmon();
+    const bool fReindex = Blocks::DB::instance()->isReindexing();
+    logDebug(Log::Net) << "received:" << SanitizeString(strCommand) << "bytes:" << vRecv.size() << "peer:" << pfrom->id;
+    if (mapArgs.count("-dropmessagestest") && GetRand(atoi(mapArgs["-dropmessagestest"])) == 0)
+    {
+        LogPrintf("dropmessagestest DROPPING RECV MESSAGE\n");
+        return true;
     }
 
+    const bool xthinEnabled = IsThinBlocksEnabled();
+
+    NetworkMessage *networkMessage = nullptr;
+    if (strCommand == NetMsgType::VERSION)
```

Comment:
You moved curly braces onto the same line as the if in another part, I'm a bit surprised you don't do that here too :)

I prefer curly braces on the same line as the if() statement.

Reply:
ok, me too :) (coming from java)